source/docs/software/advanced-controls/introduction/approaches-to-ctrl-sys-design.rst

```diff
@@ -0,0 +1,70 @@
+
+:doc:`WPILib's system identification tool </docs/software/pathplanning/system-identification/introduction>` supports some common FRC mechanisms, including drivetrain. It deploys its own code to the robot to exercise the mechanism, record data, and derive relevant constants. These constants can be used to build a more accurate model of the system behavior.
+
+
+Correcting for Unknowns with Feedback
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Even with lots of experiments and study, it is not possible to know every force that will be exerted on a robot's mechanism with great detail. For example, in a flywheel shooter, the timing and exact forces associated with a ball being put through the mechanism are not easy to define accurately. For another example, consider the fact that gearboxes can be greased to reduce friction, but throw off grease over time and get more friction. This is a complex process to model well.
+
+In cases where not everything can be known, incorporating sensors and *feedback* control is necessary to correct for unknowns in the system's behavior.
+
+If the unmodeled, unknown behaviors of the system are big enough to prevent a mechanism from working well, feedback control will be necessary to control the system.
+
+
+Incorporating Feedback
+----------------------
+
+Let's say we are controlling :doc:`a DC Motor </docs/software/pathplanning/system-identification/introduction>`. With just a mathematical model and knowledge of all the current states of the system (i.e., angular velocity), we can predict all future states given the future voltage inputs. Why then do we need feedback control? If the system is disturbed in any way that isn't modeled by our equations, like a load was applied, or voltage sag in the rest of the circuit caused the commanded voltage to not match the applied voltage, the angular velocity of the motor will deviate from the model over time.
+
+To combat this undesired effect, we can take measurements of the system and the environment to detect this deviation and account for it. For example, we could measure the current position and estimate an angular velocity from it. We can then give the motor corrective commands as well as steer our model back to reality. This feedback allows us to account for uncertainty and be robust to it.
+
+Feedback-Only Techniques
+------------------------
+
+In many controls textbooks, you may see a set of techniques which rely on feedback control only. These are very common in industry, and works well in many cases, especially when the underlying system behavior is not easy to model.
+
+For most FRC mechanisms, these feedback-only techniques may produce good results in some cases. However, there will be limits to this success. For most FRC use cases, it is recommended to include feedforward.
```

Comment:
With the exception of flywheels and one complicated multi-jointed arm, feedback only has been sufficient in my experience. This seems to undersell it's usefulness.

Reply:
It works if you throw enough motor power at it, but it's really not that hard to end up in a situation where the mechanism you have does not work without feedforward.

Reply:
However, NEOs and Falcons make a lot of subsystems overpowered nowadays.

Reply:
The turret demo is currently awfully overpowered.  This proves the point well for "feedback only sometimes works," though we might want to rescale the control effort plot...